subsys/disk/disk_access_stm32_sdmmc.c

```diff
@@ -108,12 +120,38 @@ static int stm32_sdmmc_access_init(struct disk_info *disk)
 		return err;
 	}
 
+	/* Initialize semaphore */
+	k_sem_init(&priv->sem, 1, 1);
+	k_sem_init(&priv->sync, 0, 1);
+
+	/* Run IRQ init */
+	priv->irq_config(dev);
+
+	priv->hsd.Init.ClockEdge = SDMMC_CLOCK_EDGE_RISING;
+	priv->hsd.Init.ClockPowerSave = SDMMC_CLOCK_POWER_SAVE_DISABLE;
+	priv->hsd.Init.BusWide = SDMMC_BUS_WIDE_4B;
+	priv->hsd.Init.HardwareFlowControl = SDMMC_HARDWARE_FLOW_CONTROL_DISABLE;
+	priv->hsd.Init.ClockDiv = 1;
+	priv->hsd.Init.TranceiverPresent = SDMMC_TRANSCEIVER_NOT_PRESENT;
```

Comment:
This part is breaking compilation on stm32f746g_disco

Reply:
This and then F7 has no build-in DMA. I disabled DMA mode for anything except H7 and even there it is Kconfig option. Driver still works on H7 :-)